packages/idyll-components/src/generateHeaders.js

```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+const generateId = (headerText = '') => {
+  return headerText
+    .toString()
+    .replace(/\s+/g, '-')
+    .toLowerCase();
+};
+
+const GenerateHeaders = props => {
+  const { size, children = [] } = props;
+  const headerText = children.join('');
+  const generatedId = generateId(headerText);
+  const HeaderTag = `h${size}`;
+  return <HeaderTag id={generatedId}>{headerText}</HeaderTag>;
```

Comment:
This should pass other properties (e.g. `className`) into the returned component, and should only use the generated ID if one isn't provided by the user.

Reply:
Done!!